### PR TITLE
adding base integration test system

### PR DESCRIPTION
**Pull request by RMI78**
MR for the integration test system. This is the base system with the documentation, tests will come once this MR will be merged.

As of today, I can and will provide integration tests for the following modules:
- backup 
- wp_enum 
- drupal_enum
- brute_login_form 
- cookieflags 
- crlf 
- csp 
- exec 
- file 
- htaccess 
- http_headers 
- xss 
- shellshock 
- methods 
- timesql 
- wapp
- permanentxss
- log4shell
- redirect
- ssrf

As well as a test for the crawler and its different way to handle redirections, plus a test for every authenticating methods it does. 

You can look up in the documentation to see how it works

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -43,3 +43,8 @@ pip-delete-this-directory.txt
 
 ### Pycharm ###
 .idea/
+
+### integration test system ###
+tests/integrations/.dump_diff_file.txt
+tests/integrations/.test/
+tests/integrations/docker-compose.solo_setup.yml
```
Comment (Slokilla) [inline .gitignore:50]:
```suggestion

```

File: tests/integration/check.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+# 1 assertion per target, one or more targets per module,
+# assertions can be found in the same directory than this file
+declare -a assertions
+mapfile -t assertions < <(find . -name "*.json")
+if [[ ${#assertions[@]} -eq 0 ]]; then 
+    die "Error: No assertions found in module $MODULE_NAME"
+fi
+
+# Since the assertions and the target reports must have the same name 
+# (except their extention names, .txt and .out) we extract the targets names
```
Comment (Slokilla) [inline tests/integration/check.sh:37]:
```suggestion
# (except their extention names, .json and .out) we extract the targets names
```
no ?

File: tests/integration/check.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+# Wapiti may not detect some targets or it can be due to various misconfigurations
+# so we check the size of each array to be sure they match
+if [[ ${#assertions[@]} -ne ${#targets_name[@]} || ${#targets_name[@]} -ne ${#outputs[@]} ]] ; then
+    die "Error: different number of reports/assertion files, found ${#outputs[@]} outputs for ${#assertions[@]} assertions"
+fi
+
+EXIT_CODE=0
+# Comparing outputs and assertions is a 2-steps process:
+for i in "${!outputs[@]}"; do
+
```
Comment (Slokilla) [inline tests/integration/check.sh:66]:
```suggestion

```

File: tests/integration/run.sh [hunk 4]
```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# List of the currently working tests
```
Comment (Slokilla) [inline tests/integration/run.sh:3]:
```suggestion
# List of modules to be tested
```
is more accurate ?

File: tests/integration/wapiti/Dockerfile.integration [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+RUN apt-get -y update &&\
+  apt-get -y install --no-install-recommends\
+  python3 python3-pip python3-setuptools ca-certificates &&\
+  apt-get -y clean &&\
+  apt-get -y autoremove &&\
+  rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* &&\
+  truncate -s 0 /var/log/*log
+
+COPY . .
+
```
Comment (Slokilla) [inline tests/integration/wapiti/Dockerfile.integration:17]:
```suggestion

```

**Review by Slokilla — CHANGES_REQUESTED** [review]

**Review by fwininger — APPROVED** [review]

**Review by bretfourbe — APPROVED** [review]

**Pull request merged**